react/features/mobile/polyfills/querySelectorPolyfill.js

```diff
@@ -0,0 +1,281 @@
+ */
+function _handleAttributeSelector(node, selector, findFirst) {
+    const { tagName } = _parseSelector(selector); // Just to get tagName for optimization
+
+    // Handler's job: find the right elements to search
+    const elementsToCheck = tagName
+        ? Array.from(node.getElementsByTagName(tagName))
+        : Array.from(node.getElementsByTagName('*'));
+
+    // Common helper does the matching
+    return _filterAndMatchElements(elementsToCheck, selector, findFirst);
+}
+
+/**
+ * Internal function that implements the core selector matching logic for both
+ * querySelector and querySelectorAll. Supports :scope pseudo-selector, direct
+ * child selectors, and common CSS selectors.
+ *
+ * @param {Node} node - The Node which is the root of the tree to query.
+ * @param {string} selector - The CSS selector to match elements against.
+ * @param {boolean} findFirst - If true, return after finding the first match.
+ * @returns {Element[]|Element|null} - Array of Elements for querySelectorAll,
+ * single Element or null for querySelector.
+ */
+function _querySelectorInternal(node, selector, findFirst = false) {
+    // Normalize whitespace around > combinators first
+    const normalizedSelector = selector.replace(/\s*>\s*/g, '>');
+
+    // Handle :scope pseudo-selector
+    if (normalizedSelector.startsWith(':scope')) {
+        const scopeResult = _handleScopeSelector(node, normalizedSelector, findFirst);
+
+        if (scopeResult !== null) {
+            return scopeResult;
+        }
+    }
+
+    // Handle nested > selectors (direct child combinators)
+    if (normalizedSelector.includes('>')) {
+        return _handleDirectChildSelectors(node, normalizedSelector, findFirst);
+    }
+
+    // Fast path: simple tag name
+    if (normalizedSelector === '*' || /^[a-zA-Z][\w-]*$/.test(normalizedSelector)) {
+        return _handleSimpleTagSelector(node, normalizedSelector, findFirst);
+    }
+
+    // Attribute selector: tagname[attr="value"] or
+    // tagname[attr1="value1"][attr2="value2"] (with optional wildcard namespace)
+    if (normalizedSelector.match(/^([a-zA-Z][\w-]*)?(\[(?:\*\|)?([^=\]]+)=["']?([^"'\]]+)["']?\])+$/)) {
+        return _handleAttributeSelector(node, normalizedSelector, findFirst);
+    }
+
+    // Unsupported selector - throw SyntaxError to match browser behavior
+    throw new SyntaxError(`Failed to execute 'querySelector${
+        findFirst ? '' : 'All'}' on 'Element': '${selector}' is not a valid selector.`);
+}
+
+/**
+ * Implements querySelector functionality using the shared internal logic.
+ * Supports the same selectors as querySelectorAll but returns only the first match.
+ *
+ * @param {Node} node - The Node which is the root of the tree to query.
+ * @param {string} selectors - The CSS selector to match elements against.
+ * @returns {Element|null} - The first Element which matches the selector, or null.
+ */
+export function querySelector(node, selectors) {
```

Comment:
Can we have some basic tests for this? Also, did you notice any chance in speed when processing meeting joins?